### ISSUE TITLE
writes in Show format ?

### ISSUE THREAD

**Issue by simonmichael** [valderman/tini#1]
Thanks for this nifty lib, which seems just right - small and simple enough that I couldn't screw it up. But...

I have
```haskell
data SavedState = SavedState
  { sshighscore :: Score
  } deriving (Generic, Show, Eq)

instance Configurable SavedState where
  defaultConfig = SavedState { sshighscore = 0 }

-- Where to save persistent state.
saveFile :: IO FilePath
saveFile = (</> progname <.> "save") <$> getXdgDirectory XdgData progname

-- Write the game's saveable state to the filesystem, if possible.
-- Catch any error and return the message.
gameSave :: Game -> IO (Maybe String)
gameSave Game{gsaved} = handle (\(e::IOException) -> return $ Just $ show e) $ do
  f <- saveFile
  createDirectoryIfMissing True $ takeDirectory f
  writeConfigFile f gsaved
  return Nothing    
```

writeConfigFile seems to be writing data in Show format (and readConfigFile silently fails to read that):
```
Ini {unIni = [(SN "",[(KeyPart "sshighscore","1")])]}
```

This seems contrary to the README, why is it doing that ?

**Comment by simonmichael** [comment]
Perhaps it's because I added the Show instance ? That's something I need for debugging etc.

**Comment by valderman** [comment]
Thanks for spotting this, `writeIniFile` stupidly used `show` instead of `showIni` to serialize the config before writing it out. Fixed in v0.1.0.1 which will be on Hackage any minute now.

I also thought it was small and simple enough to be impossible to screw up, but I did. :P

**Comment by simonmichael** [comment]
Thanks! :-)
